### PR TITLE
CC Automation/Comp Eval coverage - promote CCV with specific registry_name

### PR DESCRIPTION
**Pull request by sambible**
CC Automation/Comp Eval for CV work for BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2153523

### CODE REVIEW

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_contentview.py -k 'test_ccv_promote_registry_name_change'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6154
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_contentview.py -k test_ccv_promote_registry_name_change --external-logging
Test Result : ================= 87 deselected, 1 error in 786.88s (0:13:06) ==================
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6287
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_contentview.py -k test_ccv_promote_registry_name_change --external-logging
Test Result : =========== 87 deselected, 3 warnings, 1 error in 705.16s (0:11:45) ============
```

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_contentview.py -k 'test_ccv_promote_registry_name_change'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6360
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_contentview.py -k test_ccv_promote_registry_name_change --external-logging
Test Result : =========== 87 deselected, 3 warnings, 1 error in 784.75s (0:13:04) ============
```

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_contentview.py -k 'test_ccv_promote_registry_name_change'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6363
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_contentview.py -k test_ccv_promote_registry_name_change --external-logging
Test Result : =========== 1 passed, 87 deselected, 4 warnings in 854.96s (0:14:14) ===========
```

**Review by damoore044 — APPROVED** [review]
ack nice work !

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_contentview.py -k 'test_ccv_promote_registry_name_change'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 76
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_contentview.py -k test_ccv_promote_registry_name_change --external-logging
Test Result : ========== 1 passed, 88 deselected, 97 warnings in 771.90s (0:12:51) ===========
```

File: tests/foreman/api/test_contentview.py [hunk 1]
```diff
@@ -278,6 +279,63 @@ def test_positive_add_sha512_rpm(self, content_view, module_org, module_target_s
...
+            organization=module_sca_manifest_org,
+            registry_name_pattern='<%= repository.name %>',
+        ).create()
+        lce2 = module_target_sat.api.LifecycleEnvironment(
+            organization=module_sca_manifest_org,
+            registry_name_pattern='<%= lifecycle_environment.label %>/<%= repository.name %>',
+        ).create()
+        version = composite_cv.version[0].read()
+        assert version.promote(data={'environment_ids': lce1.id})
+        assert version.promote(data={'environment_ids': lce2.id})
```
Comment (vsedmik) [inline tests/foreman/api/test_contentview.py:337]: non-blocking
```suggestion
        assert 'success' in version.promote(data={'environment_ids': lce1.id})['result']
        assert 'success' in version.promote(data={'environment_ids': lce2.id})['result']
```

**Review by vsedmik — APPROVED** [review]
Looks nice. @sambible could you rebase, resolve the conflict and push once again?

File: tests/foreman/api/test_contentview.py [hunk 2]
```diff
@@ -278,6 +279,63 @@ def test_positive_add_sha512_rpm(self, content_view, module_org, module_target_s
...
+        """Testing CCV promotion scenarios where the registry_name has been changed to some
+        specific value.
+
+        :id: 41641d4a-d144-4833-869a-284624df2410
+
+        :steps:
+
+            1) Sync a RH Repo
+            2) Create a CV, add the repo and publish it
+            3) Create a CCV and add the CV to it, then publish it
```
Comment (vijaysawant) [inline tests/foreman/api/test_contentview.py:293]: should be content view version
```suggestion
            3) Create a CCV and add the CV version to it, then publish it
```

**Review by vijaysawant — CHANGES_REQUESTED** [review]
please look over steps.

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_contentview.py -k 'test_ccv_promote_registry_name_change'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6753
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_contentview.py -k test_ccv_promote_registry_name_change --external-logging
Test Result : ========== 1 passed, 88 deselected, 97 warnings in 804.24s (0:13:24) ===========
```

**Review by vsedmik — APPROVED** [review]
ACK

**Pull request merged**